homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -78,6 +80,55 @@
     required=True,
 )
 
+
+def _above_greater_than_below(configs: list) -> list:
+    config = configs[0]
+    if config.get(CONF_PLATFORM, None) != CONF_NUMERIC_STATE:
+        return configs
+    above = config.get(CONF_ABOVE, None)
+    below = config.get(CONF_BELOW, None)
+    if above is None and below is None:
+        raise vol.Invalid(
+            "For bayesian numeric state at least one of 'above' or 'below' must be specified."
+        )
+    if above is not None and below is not None:
+        if above > below:
+            raise vol.Invalid(
+                "For bayesian numeric state 'above' must be less than 'below'."
+            )
+    return configs
+
+
+def _no_overlaping(configs: list[dict]) -> list[dict]:
+    numeric_configs = [
+        config
+        for config in configs
+        if config.get(CONF_PLATFORM, None) == CONF_NUMERIC_STATE
+    ]
+    if len(numeric_configs) < 2:
+        return configs
+    d: dict[str, list[tuple]] = {}
+    index = 0
+    for index, config in enumerate(numeric_configs):
+        above = config.get(CONF_ABOVE, -math.inf)
+        below = config.get(CONF_BELOW, math.inf)
+        entity_id: str = str(config.get(CONF_ENTITY_ID))
+        d.setdefault(entity_id, [])
+        d[entity_id].append((index, above, below))
+
```

Comment:
I would use named tuples to make this section more readable

Reply:
Fixed in the next commit. I learnt a new python thing! Thanks :)